test/action_view/integration_test.rb

```diff
@@ -35,6 +35,15 @@ class IntegrationTest < ActionDispatch::IntegrationTest
     HTML
   end
 
+  test "rendering component with content_for" do
+    get "/content_for"
+    assert_response :success
+
+    assert_includes response.body, "<h1>Hi!</h1>"
+    assert_includes response.body, "<p>Have a great day.</p>"
+    assert_includes response.body, "<h3>Bye!</h3>"
```

Comment:
I'd like to assert all the HTML but there are whitespace and new line differences that are a pain to account for.